config/config-api/src/com/thoughtworks/go/config/remote/ConfigRepoConfig.java

```diff
@@ -71,6 +79,16 @@ public String getConfigProviderPluginName() {
         return configProviderPluginName;
     }
 
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        if(StringUtil.isBlank(id))
+            id = null;
```

Comment:
We don't do that with elastic agent and auth plugins. Users are expected to provide an identifier.